db/event_manager_test.go

```diff
@@ -748,5 +724,26 @@ func TestUnavailableWebhook(t *testing.T) {
 		em.RaiseDocumentChangeEvent(body, "", channels)
 	}
 
-	time.Sleep(50 * time.Millisecond)
+}
+
+// asserts that the number of items seen in the channel within the specified time limit is the same as the expected value.
+// WARNING: This function will drain the channel of items!
+func assertChannelLengthWithTimeout(t *testing.T, c chan Body, expectedLength int, timeout time.Duration) {
+	count := 0
+	for {
+		if count >= expectedLength {
+			// Make sure there are no additional items on the channel after a short wait.
+			// This avoids relying on the longer timeout value for the final check.
+			time.Sleep(timeout / 100)
```

Comment:
Nice catch to notice that we also want to handle the case where more events show up in the channel than expected.  It's unfortunate that we're stuck still using a sleep here, but I agree that there isn't a better way to handle with the code as it stands today.  An enhancement to add a 'EventManager.Close()', that would drain events then close the channel, is probably the missing piece.  That's out of the scope of this PR, though.